app/components/close_help_post.py

```diff
@@ -123,15 +128,73 @@ async def close_post(
         return
 
     await interaction.response.defer(ephemeral=True)
+    followup = "Post closed."
 
     desired_tag_id = config.HELP_CHANNEL_TAG_IDS[tag]
     await post.add_tags(next(tag for tag in help_tags if tag.id == desired_tag_id))
 
     if title_prefix is None:
         title_prefix = f"[{tag.upper()}]"
-    await post.edit(name=f"{title_prefix} {post.name}")
+
+    try:
+        await post.edit(name=f"{title_prefix} {post.name}")
+    except discord.HTTPException as e:
+        # Re-raise if it's not because the new post title was invalid.
+        if e.status != INVALID_REQUEST_DATA or e.code != INVALID_FORM_BODY:
+            raise
+
+        # HACK: there does not appear to be any way to get the actual error
+        # without parsing the returned string or using a private field.  This
+        # is likely a limitation of discord.py, as the Discord API
+        # documentation mentions:
+        #     Some of these errors may include additional details in the form
+        #     of Error Messages provided by an errors object.
+        # in https://discord.com/developers/docs/topics/opcodes-and-status-codes#json
+        # Both approaches are going to be tried... here be dragons.
+        try:
+            # The format of e._errors that is being parsed below is
+            # {
+            #     "name": {
+            #         "_errors": [
+            #             {
+            #                 "code": "BASE_TYPE_MAX_LENGTH",
+            #                 "message": "Must be 100 or fewer in length.",
+            #             }
+            #         ]
+            #     }
+            # }
+            # This tries to make as few assumptions of the keys of the
+            # dictionary as possible, as the field is private.
+            returned_error = next(iter(next(iter(e._errors.values())).values()))[0][  # noqa: SLF001
+                "message"
+            ]
+        except (StopIteration, IndexError):
+            try:
+                # Parsing something of the form:
+                # "Invalid Form Body\nIn name: Must be 100 or fewer in length."
+                returned_error = (
+                    str(e).splitlines()[1].removeprefix("In name: ").strip()
+                )
+            except IndexError:
+                returned_error = None
+        if not returned_error:
+            followup = "Unable to change the post title."
+        else:
+            # Try to fix up some grammar and punctuation.
+            returned_error = returned_error[0].lower() + returned_error[1:]
+            if returned_error[-1] not in ".?!":
+                returned_error += "."
+            returned_error = returned_error.replace("must be", "must have been")
+            followup = f"Unable to change the post title as it {returned_error}"
```

Comment:
I'd say, since we know what the original `returned_error` will generally be, maybe this should be hard-coded instead? So something like this:
```suggestion
        if returned_error == TOO_LONG_ERROR:
            followup = "Unable to change the post title because it would become >100 characters long."
        else:
            followup = "Unable to change the post title."
```
or
```suggestion
        if returned_error == TOO_LONG_ERROR:
            reason = " because it would become >100 characters long"
        else:
            reason = ""
        followup = f"Unable to change the post title{reason}."
```

Reply:
> `returned_error == TOO_LONG_ERROR`

Not sure where I can get a `TOO_LONG_ERROR` from, but I could do a `if returned_error:` instead similar to what I had originally done.

Reply:
My suggestion assumed you'd throw a
```py
TOO_LONG_ERROR = "Must be 100 or fewer in length."
```
in there, I should've been more clear. :sweat_smile:

Also, I think it would be handy to forcefully crash this function if there's a different error than just "name too long". It would help with discovery both locally and on Sentry, as right now it just gets silenced with a generic followup message.

Reply:
> My suggestion assumed you'd throw a […] in there, I should've been more clear. 😅

I'd personally prefer to do something like `returned_error.contains("or fewer in length")`, though I agree that harms readability more than having the whole string there.

> I think it would be handy to forcefully crash this function if there's a different error than just "name too long"

Definitely, but I also don't trust discord to keep their strings straight[^1]…  That is, if they change the limit or something, our check is going to fail.  Though I'm fine with doing that and updating this code when we get this issue again on Sentry.

[^1]: Their docs specifically say this is an additional message, not something to be parsed.  So far I've only parsed the extra stuff discord.py adds on around Discord's message, as well as some best-effort string replacements.

Reply:
For now I'm going to rethrow if it's not a name that's too long, and do a `"or fewer in length" in returned_error`, let me know your opinion and I'll switch it out to whatever we decide on.

Reply:
> For now I'm going to [...] do a `"or fewer in length" in returned_error`

Fine by me

Reply:
Done.